lib/go-tc/cdn_lock.go

```diff
@@ -0,0 +1,33 @@
+package tc
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import "database/sql"
+
+// AcmeAccount is the information needed to access an account with an ACME provider.
+type CdnLock struct {
+	UserName    string    `json:"userName" db:"user_name"`
+	CdnName     string    `json:"cdnName" db:"cdn_name"`
+	LastUpdated TimeNoMod `json:"lastUpdated" db:"last_updated"`
```

Comment:
you should probably just add an optional Message. Pretty sure if somebody locks a CDN, they are going to have to provide a reason...